src/box/lua/config/utils/schema.lua

```diff
@@ -1492,6 +1492,88 @@ end
 
 -- }}} <schema object>:pairs()
 
+-- {{{ <schema object>:jsonschema()
+
+local json_scalars = {
+    string = {jsonschema = {type = 'string'}},
+    number = {jsonschema = {type = 'number'}},
+    ['string, number'] = {jsonschema = {type = {'string', 'number'}}},
+    ['number, string'] = {jsonschema = {type = {'string', 'number'}}},
+    integer = {jsonschema = {type = 'integer'}},
+    boolean = {jsonschema = {type = 'boolean'}},
+    any = {jsonschema = {}}
+}
+
+-- This function sets common fields such as `description`, `default`, and `enum`
+-- for a given schema object. These fields are typical in JSON schemas.
+local function set_common_fields(res, schema)
+    if schema.description ~= nil then
+        res.description = schema.description
+    end
+    res.default = schema.default
+    res.enum = schema.allowed_values
+    return setmetatable(res, {
+        __serialize = 'map',
+    })
+end
+
+local function traverse_impl(schema)
+    local schema_type = schema.type
+
+    if is_scalar(schema) then
+        -- If the schema is a scalar type (string, number, etc.)
+        -- copy the corresponding JSON schema.
+        local scalar_copy = table.copy(json_scalars[schema_type].jsonschema)
+        return set_common_fields(scalar_copy, schema)
+
+    elseif schema_type == 'record' then
+        local properties = {}
+        for field_name, field_def in pairs(schema.fields) do
+            properties[field_name] = traverse_impl(field_def)
+        end
+        return set_common_fields({
+            type = 'object',
+            properties = properties,
+            additionalProperties = false
+        }, schema)
+
+    elseif schema_type == 'map' then
+        assert(schema.key.type == 'string')
+        return set_common_fields({
+            type = 'object',
+            additionalProperties = traverse_impl(schema.value)
+        }, schema)
+
+    elseif schema_type == 'array' then
+        local res = {
+            type = 'array',
+            items = traverse_impl(schema.items),
+        }
+        if schema.validate == validate_no_repeat then
+            res.uniqueItems = true
+        end
+        return set_common_fields(res, schema)
+
+    else
+        error('Unsupported schema type: ' .. tostring(schema_type))
```

Comment:
It shouldn't occur if the schema is constructed using schema node constructors from this module. So, I would just `assert(false)` here in other schema traversing functions.

Reply:
Replaced with `assert(false)`.